### ISSUE TITLE
Filename containing multiple words and dots fail to get served

### ISSUE THREAD

**Issue by tiptronic** [timberio/gitdocs#30]
Uncommon filenames don't link
e.g. "Props vs. State.md"

Not sure what the correct way to act on these files is: but at least there should be an error thrown...

**Comment by etimberg** [comment]
Some investigation says that the space is responsible. If I have `Props_vs._State.md` it works fine, but `Props vs. State.md` does not.

**Comment by etimberg** [comment]
Also, this only seems to affect `gitdocs serve`. If you do `gitdocs build` and then serve the `docs-dist` folder it works as expected.

**Comment by tiptronic** [comment]
Not for me... (MacOS X 10.3.2)

<img width="709" alt="screenshot 2017-12-20 17 55 49" src="https://user-images.githubusercontent.com/80752/34218733-4d4a7ff6-e5af-11e7-9c22-c2ae052bdfcf.png">

**Comment by etimberg** [comment]
Weird, I served the build output using the following on OSX 10.11.6.

```
npm install -g serve
serve docs-dist -p 3000
```

I tracked down the issue with `gitdocs serve` to an issue in `react-static`. PR sent in https://github.com/nozzle/react-static/pull/250 to fix it.

**Comment by tiptronic** [comment]
? - I used:

gitdocs serve docs/dist/

**Comment by etimberg** [comment]
Ah, so the `gitdocs serve` command always runs the `react-static` dev server underneath which has the problem fixed by the PR linked above.

Serving the dist folder independently was meant to test serving deployed files from somewhere else to see if `gitdocs` was generating incorrect output or if the issue was isolated to the `gitdocs serve` command.

I'm going to try and write up some documentation on all the CLI commands this afternoon.

**Comment by tiptronic** [comment]
Jep - you're right. Using global 'serve' works fine (except bug #31).

Is there any chance, we get this to also run from the file-system? (I tend to click everything on my disc, instead serving 😉)

**Comment by etimberg** [comment]
definitely. I think we can make `gitdocs serve` take an optional path. if specified, it serves that directory, else it does the current behaviour. Would that work for you?

**Comment by tiptronic** [comment]
@etimberg if you type just  ```gitdocs``` into your terminal, it states that it does so already:

```bash
 gitdocs serve [path]              serve
 gitdocs build [output] [version]  build
 gitdocs init                      init
```

**Comment by etimberg** [comment]
oh ..... cool, even better then! I'll see if I can quickly implement it.

**Comment by etimberg** [comment]
@tiptronic the path argument is now used in #32
